examples/pylab_examples/mri_demo.py

```diff
@@ -9,8 +9,7 @@
 
 # Data are 256x256 16 bit integers
 dfile = cbook.get_sample_data('s1045.ima.gz')
-im = np.fromstring(dfile.read(), np.uint16).astype(float)
-im.shape = (256, 256)
+im = np.fromstring(dfile.read(), np.uint16).reshape((256, 256))
```

Comment:
Ah, unfortunate. Based on `np.fromfile` seeming to need `fileno` (since it doesn't work with `BytesIO`), it looks like it tries to stream from the fd directly, which `GzipFile` points at the underlying file object. In a way, it's a half-a-bug in both coming together to make one.